### PR TITLE
chore: logging improvements

### PR DESCRIPTION
**Pull request by dpc**

### CODE REVIEW

File: fedimint-client/src/sm/executor.rs [hunk 1]
```diff
@@ -279,7 +279,7 @@ impl Executor {
         );
 
         let task_runner_inner = self.inner.clone();
-        let _handle = self.inner.client_task_group.spawn("state machine executor", |task_handle| async move {
```
Comment (dpc) [inline fedimint-client/src/sm/executor.rs:282]: Too long

File: fedimint-client/src/sm/executor.rs [hunk 2]
```diff
@@ -511,10 +511,11 @@ impl ExecutorInner {
                     debug!(
                         target: LOG_CLIENT_REACTOR,
                         operation_id = %state.operation_id(),
-                        "State machine trigger function complete. Starting transition function.",
+                        "Triggered state transition",
                     );
-                    let span = tracing::info_span!(
-                        "state_machine_transition",
```
Comment (dpc) [inline fedimint-client/src/sm/executor.rs:517]: Too long

File: fedimint-client/src/sm/executor.rs [hunk 3]
```diff
@@ -532,12 +533,10 @@ impl ExecutorInner {
                             async move {
                                 debug!(
                                     target: LOG_CLIENT_REACTOR,
-                                    operation_id = %state.operation_id(),
```
Comment (dpc) [inline fedimint-client/src/sm/executor.rs:535]: Redundant with the `debug_span` above

File: fedimint-core/src/runtime.rs [hunk 4]
```diff
@@ -126,6 +127,7 @@ mod r#impl {
     where
         F: Future<Output = T> + 'static,
     {
+        let span = tracing::debug_span!(target: LOG_RUNTIME, "spawn", task = name);
```
Comment (dpc) [inline fedimint-core/src/runtime.rs:130]: Missing

File: fedimint-core/src/task.rs [hunk 5]
```diff
@@ -172,11 +172,7 @@ impl TaskGroup {
         Fut: Future<Output = R> + MaybeSend + 'static,
         R: MaybeSend + 'static,
     {
-        use tracing::{info_span, Instrument, Span};
-
         let name = name.into();
-        // new child span of current span
-        let span = info_span!(parent: Span::current(),"task", name);
```
Comment (dpc) [inline fedimint-core/src/task.rs:179]: `parent` leads to a lot of not very useful nested spawns, and the whole thing is redudant with with the `runtime::spawn` doing the same thing.

File: fedimint-core/src/task/jit.rs [hunk 6]
```diff
@@ -90,10 +90,7 @@ where
     where
         Fut: Future<Output = std::result::Result<T, E>> + 'static + MaybeSend,
     {
-        let handle = crate::runtime::spawn(
-            &format!("JitTry {} value", std::any::type_name::<T>()),
```
Comment (dpc) [inline fedimint-core/src/task/jit.rs:94]: Waay too long. Like waaay, waay too long.

**Review by bradleystachurski — APPROVED** [review]

**Pull request merged**